src/schnorr/ed25519.rs

```diff
@@ -0,0 +1,56 @@
+use ic_ed25519::{DerivationIndex, DerivationPath, PublicKey};
+
+use pretty_assertions::assert_eq;
+
+use crate::test_vector::{ChainCode, SerializedDerivationPath, load_test_vectors};
+
+/// Converts a derivation path from the test vectors into the equivalent type in the `ic_secp256k1` crate.
+impl From<SerializedDerivationPath> for DerivationPath {
+    fn from(path: SerializedDerivationPath) -> Self {
+        Self::new(path.elements.into_iter().map(DerivationIndex).collect())
+    }
+}
+
+/// A public key together with the corresponding chain code.
+#[derive(Debug, PartialEq, Eq)]
```

Comment:
Are the `Eq`/`PartialEq` derivations used?

I guess it's harmless to define them and not use them immediately, certainly equality on this type is logically sound